src/v0/sources/shopify/transform.js

```diff
@@ -126,6 +132,27 @@ const processEvent = (inputEvent) => {
     case ECOM_TOPICS.CHECKOUTS_UPDATE:
       message = ecomPayloadBuilder(event, shopifyTopic);
       break;
+    case 'carts_create':
+    case 'carts_update':
+      /**
+       *  This Scenario handles the case same cart_Events are passed or
+       * cart events are passed within a specified time period for when useRedisDatabase is set to true
+       */
+      // eslint-disable-next-line no-case-declarations
+      const validRecord = await checkForValidRecord(inputEvent);
+
+      if (useRedisDatabase && !validRecord) {
```

Comment:
Don't we need to check `useRedisDatabase` before calling to `checkForValidRecord`?

Reply:
Yeah changed